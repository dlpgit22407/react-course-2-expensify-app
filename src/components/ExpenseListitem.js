import React from 'react';
import { Link } from 'react-router-dom';
//import { connect } from 'react-redux';
//import { removeExpense } from '../actions/expenses';

const ExpenseListitem = ({ dispatch, description, id, amount, createdAt }) => (
    <div>
        <Link to={`/edit/${id}`}>
            <h3>{description}</h3>
        </Link>
        <p>{amount} - {createdAt}</p>      
    </div>
);

export default ExpenseListitem
//export default connect()(ExpenseListitem);
