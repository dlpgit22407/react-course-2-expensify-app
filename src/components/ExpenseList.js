import React from 'react';
import { connect } from 'react-redux';
import ExpenseListitem from './ExpenseListitem';
import selectExpenses from '../selectors/expenses'; 

export const ExpenseList = (props) => (
    <div>
        {
            props.expenses.length === 0 ? (
                <p>No expeneses</p>
            ) : (
                props.expenses.map((expense) => {
                return <ExpenseListitem key={expense.id} {...expense} />;
            })
            )
        }
    </div>
);
 //{props.filters.text}

//const ConnectedExpenseList = connect((state) => {
const mapStoreToProps = (state) => {
    return {
        expenses: selectExpenses(state.expenses, state.filters)
        //expenses: state.expenses,
        //filters: state.filters
    };
};
//})(ExpenseList);

//export default ExpenseList;
//export default ConnectedExpenseList;
export default connect(mapStoreToProps)(ExpenseList)
