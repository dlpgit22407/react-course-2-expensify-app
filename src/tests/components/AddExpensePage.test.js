import React from 'react';
import Enzyme from 'enzyme';
import { shallow } from 'enzyme';
import { AddExpensePage } from '../../components/AddExpensePage';
import expenses from '../fixtures/expenses';

import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import { configure } from 'enzyme';

Enzyme.configure({
    adapter: new Adapter()
})

let addExpense, history, wrapper;

beforeEach(() => {
    addExpense = jest.fn();
    history = { push: jest.fn() };
    wrapper = shallow(<AddExpensePage addExpense={addExpense} history={history} />)
})

test('should render addexpensepage correctly', () => {
    //const onSubmit = jest.fn();
    //const history = { push: jest.fn() };
    //const wrapper = shallow(<AddExpensePage onSubmit={onSubmit} history={history} />)
    //const wrapper = shallow(<AddExpensePage onSubmit={onSubmit} history={history} />);
    expect(wrapper).toMatchSnapshot()
});

test('should handle onSubmit', () => {
    //const onSubmit = jest.fn();
    //const history = { push: jest.fn() };
    //const wrapper = shallow(<AddExpensePage onSubmit={onSubmit} history={history} />);
    wrapper.find('ExpenseForm').prop('onSubmit')(expenses[1]);
    expect(history.push).toHaveBeenLastCalledWith('/');
    expect(addExpense).toHaveBeenLastCalledWith(expenses[1])
});