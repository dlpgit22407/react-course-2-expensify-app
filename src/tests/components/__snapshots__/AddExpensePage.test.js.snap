// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render addexpensepage correctly 1`] = `
<div>
  <h1>
    This is my add expense page.
  </h1>
  <ExpenseForm
    onSubmit={[Function]}
  />
</div>
`;
